python/srf/_pysrf/src/module_registry.cpp

```diff
@@ -47,6 +50,29 @@ bool ModuleRegistryProxy::contains_namespace(srf::pysrf::ModuleRegistryProxy& se
     return srf::modules::ModuleRegistry::contains_namespace(registry_namespace);
 }
 
+std::map<std::string, std::vector<std::string>> ModuleRegistryProxy::registered_modules(ModuleRegistryProxy& self)
+{
+    return modules::ModuleRegistry::registered_modules();
+}
+
+bool ModuleRegistryProxy::is_version_compatible(ModuleRegistryProxy& self,
+                                                const std::vector<unsigned int>& release_version)
+{
+    return modules::ModuleRegistry::is_version_compatible(release_version);
+}
+
+std::shared_ptr<modules::SegmentModule> ModuleRegistryProxy::find_module(ModuleRegistryProxy& self,
+                                                                         const std::string& name,
+                                                                         const std::string& registry_namespace,
+                                                                         const std::string& module_name,
+                                                                         py::dict module_config)
+{
+    auto json_config           = cast_from_pyobject(module_config);
+    auto fn_module_constructor = modules::ModuleRegistry::find_module(name, registry_namespace);
+    auto module                = std::move(fn_module_constructor(std::move(module_name), std::move(json_config)));
+    return std::move(module);
+}
```

Comment:
I don't think I like having the find_module binding differ from its c++ counterpart both in signature and in actual function. 

What we want to do is return a function that a user can pass module name and a py::dict to. I'm not sure we've done that directly before, will need to investigate.